app/models/miq_provision_virt_workflow.rb

```diff
@@ -218,7 +218,17 @@ def available_vlans_and_hosts(options = {})
       # TODO: Use Active Record to preload this data?
       MiqPreloader.preload(hosts, :switches => :lans)
       hosts.each do |h|
-        h.lans.each { |l| vlans[l.name] = l.name if @vlan_options[:dvs] || !l.switch.shared? }
+        h.lans.each do |l|
+          next if l.switch.shared? && !@vlan_options[:dvs]
+
+          key = if l.switch.shared?
```

Comment:
The whole method or just the DVS handling?
The issue I had with just moving the DVS stuff was that by the time it got to `manageiq-providers-vmware` the regular network and the dvs network had already collided so there was only one entry in the hash.